### PR TITLE
fixing gosimple linter errors

### PR DESCRIPTION
**Pull request by ssandeep**
I will be raising multiple PRs for fixing different linter errors. This is the first one (from gosimple linter).

### CODE REVIEW

File: types/rest/rest.go [hunk 1]
```diff
@@ -199,9 +199,9 @@ func PostProcessResponse(w http.ResponseWriter, cliCtx client.Context, resp inte
 		return
 	}
 
-	switch resp.(type) {
+	switch resp := resp.(type) {
 	case []byte:
-		result = resp.([]byte)
+		result = resp
```
Comment (ssandeep) [inline types/rest/rest.go:205]: @jdkanani @anilCSE please check if this is ok. Ref: https://nanxiao.gitbooks.io/golang-101-hacks/content/posts/type-assertion-and-type-switch.html

File: types/validator-set.go [hunk 2]
```diff
@@ -793,24 +793,24 @@ func safeSubClip(a, b int64) int64 {
...
-		absOfA = -a
-	}
-
-	if absOfA > math.MaxInt64/absOfB {
-		return 0, true
-	}
-
-	return a * b, false
-}
+// func safeMul(a, b int64) (int64, bool) {
```
Comment (jdkanani) [inline types/validator-set.go:796]: Please remove this method. We don't need it.

File: file/perm_check.go [hunk 3]
```diff
@@ -8,8 +8,8 @@ import (
 )
 
 const (
-	// storing constant as this recomended as a secruity feature
-	secretPerm os.FileMode = 0600
+// storing constant as this recomended as a secruity feature
+// secretPerm os.FileMode = 0600
```
Comment (jdkanani) [inline file/perm_check.go:12]: Please remove the code.

File: cmd/heimdalld/cmd/root.go [hunk 4]
```diff
@@ -280,48 +280,48 @@ func createSimappAndExport(
...
-		persistentPeers[i] = p2p.IDAddressString(nodeKey.ID(), fmt.Sprintf("%s:%d", hostnameOrIP(i), 26656))
-	}
+// // populate persistent peers in config
+// func populatePersistentPeersInConfigAndWriteIt(config *cfg.Config) {
+// 	persistentPeers := make([]string, totalValidators())
+// 	for i := 0; i < totalValidators(); i++ {
+// 		config.SetRoot(nodeDir(i))
+// 		nodeKey, err := p2p.LoadNodeKey(config.NodeKeyFile())
+// 		if err != nil {
+// 			return
```
Comment (jdkanani) [inline cmd/heimdalld/cmd/root.go:310]: Please remove these comments. We don't need it.

File: x/sidechannel/keeper/keeper.go [hunk 5]
```diff
@@ -194,13 +196,14 @@ func (k Keeper) IterateValidatorsAndApplyFn(ctx sdk.Context, f func(uint64, []*a
 
 		var height uint64
 		buf := bytes.NewBuffer(heightBytes)
-		binary.Read(buf, binary.BigEndian, &height)
+
+		if err := binary.Read(buf, binary.BigEndian, &height); err != nil {
+			k.Logger(ctx).Error("Error in binary.Read")
```
Comment (jdkanani) [inline x/sidechannel/keeper/keeper.go:201]: We should return after the log event.
```suggestion
			k.Logger(ctx).Error("Error in binary.Read")
			return
```

File: x/sidechannel/module.go [hunk 6]
```diff
@@ -172,7 +172,9 @@ func (am AppModule) BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock) {
 		}
 
 		// set validators for height
-		am.keeper.SetValidators(ctx, uint64(height), validators)
+		if err := am.keeper.SetValidators(ctx, uint64(height), validators); err != nil {
+			// TODO: How to handle this?
```
Comment (jdkanani) [inline x/sidechannel/module.go:176]:
```suggestion
        panic(err)
```

File: x/staking/keeper/keeper.go [hunk 7]
```diff
@@ -463,7 +463,9 @@ func (k *Keeper) Slash(ctx sdk.Context, valSlashingInfo hmTypes.ValidatorSlashin
 	validator.Jailed = valSlashingInfo.IsJailed
 
 	// add updated validator to store with new key
-	k.AddValidator(ctx, validator)
+	if err := k.AddValidator(ctx, validator); err != nil {
+		k.Logger(ctx).Error("Error calling AddValidator")
```
Comment (jdkanani) [inline x/staking/keeper/keeper.go:467]:
```suggestion
		k.Logger(ctx).Error("Error calling AddValidator")
		return err
```

File: x/staking/keeper/keeper.go [hunk 8]
```diff
@@ -485,6 +487,7 @@ func (k *Keeper) Unjail(ctx sdk.Context, valID hmTypes.ValidatorID) {
 	validator.Jailed = false
 
 	// add updated validator to store with new key
-	k.AddValidator(ctx, validator)
-	return
+	if err := k.AddValidator(ctx, validator); err != nil {
+		k.Logger(ctx).Error("Error calling AddValidator")
```
Comment (jdkanani) [inline x/staking/keeper/keeper.go:491]:
```suggestion
		k.Logger(ctx).Error("Error calling AddValidator")
		return err
```

**Review by jdkanani — CHANGES_REQUESTED** [review]

File: cmd/heimdalld/cmd/root.go [hunk 9]
```diff
@@ -50,20 +50,20 @@ import (
...
-	flagNodeDaemonHome   = "node-daemon-home"
-	flagNodeCliHome      = "node-cli-home"
-	flagNodeHostPrefix   = "node-host-prefix"
+// flagNodeDirPrefix    = "node-dir-prefix"
+// flagNumValidators    = "v"
+// flagNumNonValidators = "n"
+// flagOutputDir        = "output-dir"
+// flagNodeDaemonHome   = "node-daemon-home"
+// flagNodeCliHome      = "node-cli-home"
+// flagNodeHostPrefix = "node-host-prefix"
```
Comment (jdkanani) [inline cmd/heimdalld/cmd/root.go:62]: Please remove the commented code. We don't need it.

File: cmd/heimdalld/cmd/root.go [hunk 10]
```diff
@@ -50,20 +50,20 @@ import (
 	"github.com/maticnetwork/heimdall/types/common"
 )
 
-var logger = helper.Logger.With("module", "cmd/heimdalld")
+// var logger = helper.Logger.With("module", "cmd/heimdalld")
```
Comment (jdkanani) [inline cmd/heimdalld/cmd/root.go:53]: Remove commented code.

**Review by jdkanani — CHANGES_REQUESTED** [review]

File: x/staking/keeper/keeper.go (hunk 8 above)
Comment (ssandeep) [reply x/staking/keeper/keeper.go:491]: Unjail method doesn't have any return types, leaving this as it was earlier.

**Comment by ssandeep** [comment]
closes #450

**Review by jdkanani — APPROVED** [review]

**Pull request merged**